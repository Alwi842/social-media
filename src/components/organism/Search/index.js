import Icons from "@/components/atoms/Icons";
import { useMemo, useState } from "react";
import Router, { useRouter } from "next/router";
const Search = ({ trendingItems }) => {
  const [searchTerm, setSearchTerm] = useState("");
  const router = useRouter();
  // Filter trending items based on search input

  const filteredTrending = useMemo(() => {
    return trendingItems.filter((item) => item.toLowerCase().includes(searchTerm.toLowerCase()));
  }, [trendingItems, searchTerm]);

  return (
    <div className="w-full h-full text-white p-4 space-y-6">
      <button onClick={router.back} className="flex gap-2">
        <Icons.Back />
        Back
      </button>
      {/* Search Bar */}
      <div className="flex items-center bg-gray-800 p-2 rounded-md">
        <input
          type="text"
          placeholder="Search..."
          className="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
      </div>

      <div className="text-xl font-semibold">Trending</div>

      <div className="space-y-2">
        {/* Trending List */}
        {filteredTrending.length > 0 ? (
          filteredTrending.map((item, index) => (
            <div key={index} className="p-1 hover:bg-gray-700 rounded-md">
              {item}
            </div>
          ))
        ) : (
          <div className="text-gray-400">No trending items found</div>
        )}
      </div>
    </div>
  );
};

export default Search;
