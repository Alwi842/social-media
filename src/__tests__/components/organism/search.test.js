import { render, screen, fireEvent } from "@testing-library/react";
import Search from "@/components/organism/Search"; // Adjust the path to your file
import { useRouter } from "next/router";

// Mocking useRouter from Next.js
jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("Search Component", () => {
  const mockBack = jest.fn();
  beforeEach(() => {
    useRouter.mockReturnValue({ back: mockBack }); // Mock the back function
  });

  it("renders the component with trending items", () => {
    const trendingItems = [
      "React 18 Released",
      "JavaScript Tips",
      "Next.js vs Gatsby",
      "Web3 Development",
      "CSS Grid Layout",
      "AI in Web Development",
      "Frontend Frameworks",
      "Node.js Performance",
    ];

    // Render the Search component with the trending items
    const search = render(<Search trendingItems={trendingItems} />);
    // Ensure the "Trending" section is present
    expect(search).toMatchSnapshot();
  });

  it("filters trending items based on search input", () => {
    const trendingItems = ["React 18 Released", "JavaScript Tips", "Next.js vs Gatsby", "Web3 Development"];

    const search = render(<Search trendingItems={trendingItems} />);

    // Find the search input and simulate typing "React"
    const searchInput = search.getByPlaceholderText("Search...");
    fireEvent.change(searchInput, { target: { value: "React" } });

    // Check if only "React 18 Released" is displayed in the filtered list
    expect(search).toMatchSnapshot();
  });

  it("calls router.back when back button is clicked", () => {
    const trendingItems = ["React 18 Released", "JavaScript Tips", "Next.js vs Gatsby", "Web3 Development"];

    render(<Search trendingItems={trendingItems} />);

    // Find the back button and simulate a click
    const backButton = screen.getByText("Back");
    fireEvent.click(backButton);

    // Ensure that the mock back function was called
    expect(mockBack).toHaveBeenCalledTimes(1);
  });
});
