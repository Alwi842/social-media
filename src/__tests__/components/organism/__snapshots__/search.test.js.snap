// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component filters trending items based on search input 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full text-white p-4 space-y-6"
      >
        <button
          class="flex gap-2"
        >
          <svg
            fill="#e8eaed"
            height="24px"
            viewBox="0 -960 960 960"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
            />
          </svg>
          Back
        </button>
        <div
          class="flex items-center bg-gray-800 p-2 rounded-md"
        >
          <input
            class="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
            placeholder="Search..."
            type="text"
            value="React"
          />
        </div>
        <div
          class="text-xl font-semibold"
        >
          Trending
        </div>
        <div
          class="space-y-2"
        >
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            React 18 Released
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full text-white p-4 space-y-6"
    >
      <button
        class="flex gap-2"
      >
        <svg
          fill="#e8eaed"
          height="24px"
          viewBox="0 -960 960 960"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
          />
        </svg>
        Back
      </button>
      <div
        class="flex items-center bg-gray-800 p-2 rounded-md"
      >
        <input
          class="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
          placeholder="Search..."
          type="text"
          value="React"
        />
      </div>
      <div
        class="text-xl font-semibold"
      >
        Trending
      </div>
      <div
        class="space-y-2"
      >
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          React 18 Released
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search Component renders the component with trending items 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full text-white p-4 space-y-6"
      >
        <button
          class="flex gap-2"
        >
          <svg
            fill="#e8eaed"
            height="24px"
            viewBox="0 -960 960 960"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
            />
          </svg>
          Back
        </button>
        <div
          class="flex items-center bg-gray-800 p-2 rounded-md"
        >
          <input
            class="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
            placeholder="Search..."
            type="text"
            value=""
          />
        </div>
        <div
          class="text-xl font-semibold"
        >
          Trending
        </div>
        <div
          class="space-y-2"
        >
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            React 18 Released
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            JavaScript Tips
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            Next.js vs Gatsby
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            Web3 Development
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            CSS Grid Layout
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            AI in Web Development
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            Frontend Frameworks
          </div>
          <div
            class="p-1 hover:bg-gray-700 rounded-md"
          >
            Node.js Performance
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full text-white p-4 space-y-6"
    >
      <button
        class="flex gap-2"
      >
        <svg
          fill="#e8eaed"
          height="24px"
          viewBox="0 -960 960 960"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M640-200 200-480l440-280v560Zm-80-280Zm0 134v-268L350-480l210 134Z"
          />
        </svg>
        Back
      </button>
      <div
        class="flex items-center bg-gray-800 p-2 rounded-md"
      >
        <input
          class="bg-transparent w-full text-white placeholder-gray-400 focus:outline-none"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
      <div
        class="text-xl font-semibold"
      >
        Trending
      </div>
      <div
        class="space-y-2"
      >
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          React 18 Released
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          JavaScript Tips
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Next.js vs Gatsby
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Web3 Development
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          CSS Grid Layout
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          AI in Web Development
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Frontend Frameworks
        </div>
        <div
          class="p-1 hover:bg-gray-700 rounded-md"
        >
          Node.js Performance
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
